test/cpp/jit/test_misc.cpp

```diff
@@ -1196,6 +1198,76 @@ void testThreadLocalDebugInfo() {
   }
 }
 
+static void nestGraphIntoFallbackGraph(std::shared_ptr<Graph> graph) {
+  auto fallback =
+      createFallbackGraph(graph->block(), graph->inputs(), graph.get());
+  graph->prependNode(fallback);
+  for (size_t i = 0; i < graph->outputs().size(); i++) {
+    graph->outputs()[i]->replaceAllUsesWith(fallback->output(i));
+    fallback->output(i)->copyMetadata(graph->outputs()[i]);
+  }
+  GRAPH_DUMP("nestGraphIntoFallbackGraph:", graph);
+  EliminateDeadCode(graph);
```

Comment:
I don't think all callers of the API should have to call EliminateDeadCode after calling. It's going to make working on these passes a little more confusing if there is a bunch of dead code left around after every FallBack path is created.

Reply:
this is a helper function for this particular test; I was actually hoping that it expresses the intent better than manually removing nodes. Fixed it